File: ch8/js/a-structured-output.js
import { z } from "zod";
import { ChatAnthropic } from '@langchain/anthropic';

const joke = z.object({
  setup: z.string().describe("The setup of the joke"),
  punchline: z.string().describe("The punchline to the joke"),
});

let model = new ChatAnthropic({
  model: "claude-haiku-4-5",
  temperature: 0,
});

model = model.withStructuredOutput(joke);

const result = await model.invoke("Tell me a joke about cats");
console.log(result);

File: ch8/py/a-structured-output.py
from pydantic import BaseModel, Field
from langchain_anthropic import ChatAnthropic


class Joke(BaseModel):
    setup: str = Field(description="The setup of the joke")
    punchline: str = Field(description="The punchline to the joke")


model = ChatAnthropic(model="claude-sonnet-4-6", temperature=0)
model = model.with_structured_output(Joke)

result = model.invoke("Tell me a joke about cats")
print(result)
